.github/workflows/static_checks.yml

```diff
@@ -2,70 +2,37 @@ name: 🔍 Lint / Type Checks
 
 on:
   push:
-    branches: [main]
+    branches:
+      - main
+
   pull_request:
-    branches: [main]
+    branches:
+      - main
 
 jobs:
-  setup:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: pnpm/action-setup@v4
-        with:
-          version: 9
-      - uses: actions/setup-node@v4
-        with:
-          node-version: "20"
-          cache: "pnpm"
-      - run: pnpm i
-      - uses: actions/cache/save@v4
-        with:
-          path: |
-            node_modules
-            ~/.pnpm-store
-          key: ${{ runner.os }}-pnpm-${{ hashFiles('**/pnpm-lock.yaml') }}
```

Comment:
I don't think this is needed to cache pnpm dependencies. See the [official example](https://github.com/pnpm/action-setup#use-cache-to-reduce-installation-time).